ingestion-edge/ingestion_edge/publish.py

```diff
@@ -3,13 +3,40 @@
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
 
 from .conf import ROUTE_TABLE
+from .conf import QUEUE_PATH
 from datetime import datetime
 from flask import Blueprint, request
 from google.cloud.pubsub import PublisherClient
+from persistqueue import Queue
+from persistqueue.exceptions import Empty
 from typing import Tuple
+import threading
 
 publish = Blueprint('publish', __name__)
 client = PublisherClient()
+q = Queue(QUEUE_PATH)
+flush_lock = threading.Lock()
+
+
+@publish.route("/__flush__")
+def flush_queue() -> Tuple[str, int]:
+    if q.qsize() == 0:
+        # early return on empty queue
+        return "", 204
+    tasks_done = 0
+    with flush_lock:
+        while True:
+            try:
+                item = q.get_nowait()
+            except Empty:
+                break
+            try:
+                client.publish(**item).result()
+                q.task_done()
+                tasks_done += 1
+            except Exception:
+                return str(tasks_done), 202
```

Comment:
I'm not sure that 202 is the right status code here. 202 implies that there is a chance the work will be done in the future even if no other requests happen. From [the RFC](https://tools.ietf.org/html/rfc7231#section-6.3.3):

> The representation sent with this response ought to describe the request's current status and point to (or embed) a status monitor that can provide the user with an estimate of when the request will be fulfilled.

Looking through the status codes, I don't see any thing more appropriate than `200 OK`. Maybe if more details are needed, you could return a JSON object that has something like number of items processed/number remaining?

Reply:
how about 503? we didn't completely flush the queue, and we are temporary unable to do so. From [the RFC](https://tools.ietf.org/html/rfc7231#section-6.6.4):

> The 503 (Service Unavailable) status code indicates that the server is currently unable to handle the request due to a temporary overload or scheduled maintenance, which will likely be alleviated after some delay.

Reply:
503 seems more appropriate. It is definitely different than 202 though, since 202 implied success and 503 implies failure. Client libraries might care about that, but as long as you're consistent it shouldn't be a big deal.